### PR TITLE
6338 Use MapContext to render NavigationControls

### PR DESCRIPTION
**Pull request by godfreyyeung**
Fixes AB#6338
See  https://deck.gl/docs/api-reference/react/deckgl#react-context

image

### CODE REVIEW

File: src/components/Map/Map.tsx [hunk 1]
```diff
@@ -21,17 +24,30 @@ export const Map = ({ layers, parent }: MapProps) => {
...
   const map = (
     <DeckGL
       initialViewState={INITIAL_VIEW_STATE}
       controller={true}
       layers={layers}
       parent={parent}
+      ContextProvider={MapContext.Provider}
     >
-      <StaticMap
+      <div style={navControlStyle}>
```
Comment (TylerMatteo) [inline src/components/Map/Map.tsx:43]: Could you replace this with some style props on a `<Box>`? Or do your styles get overridden when you do it that way?

Comment (TylerMatteo) [reply src/components/Map/Map.tsx:43]: As another option, according to the docs it looks like you can pass `style` directly to `NavigationControl` so you don't need the extra div.

Comment (godfreyyeung) [reply src/components/Map/Map.tsx:43]: I tried both (Box and directly to NavigationControl), and that doesn't work. passing it to navigationcontrol directly was the first thing I tried and couldn't get that to work for the longest time. Maybe I'm missing something

File: src/components/Map/Map.tsx [hunk 2]
```diff
@@ -21,17 +24,30 @@ export const Map = ({ layers, parent }: MapProps) => {
...
   };
 
+  const navControlStyle = {
+    position: "absolute",
+    bottom: 150,
+    left: 10,
+  } as React.CSSProperties;
+
+  // MapContext is necessary for navigation controls to work.
+  // Unclear on why.
```
[the remark this answers is not in the document]
Comment (godfreyyeung) [reply src/components/Map/Map.tsx:34]: Yeah haha that's the same link I put in the PR description

File: src/components/Map/Map.tsx (hunk 1 above)
Comment (TylerMatteo) [reply src/components/Map/Map.tsx:43]: Hmm that's odd. I pulled your branch and changed it to just `<NavigationControl style={navControlStyle} />` and it seems to work, I just had to play with the value for "bottom" to get it to match up with what you have.

Comment (godfreyyeung) [reply src/components/Map/Map.tsx]: Okay good call. It wasn't working when NavigationControl was nested under ReactMapGL, but it's okay when it's outside of it

**Review by TylerMatteo — APPROVED** [review]

**Pull request merged**